projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -1116,14 +1116,9 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseDirecti
         const colLength = this.columnList.length;
         const topCols = this.columnList.filter((item) => colsArray.indexOf(item) === -1);
         if (topCols.length > 0) {
+            const columnsChanged = this.columns.length !== colLength;
```

Comment:
As far as I can see this is pretty much cleaning up the changes in this branch as they should no longer be needed, in which case I'd actually appreciate if there are no changes left here - as in just delete the else case and drop the extra `columnsChanged` variable so it matches the master source. Thanks :)

Reply:
The `columnsChanged` is regarding the issues described [https://github.com/IgniteUI/igniteui-angular/issues/12252#issuecomment-1470388757](here).
This is because once the `updateColumns` finished, the columns will be the same and `this.columns.length !== colLength` will be false.